### ISSUE TITLE
Zabbix issues with the script 

### ISSUE THREAD

**Issue by ManchM** [jack-valko/Zabbix-Meraki-Discovery#18]
Hello Jack,

I was able to fix all my issues with Zabbix, but now I'm having issues with the script
please see the attached error.

Cannot execute script: Error: cannot get URL: Timeout was reached.
at [anon] (httprequest.c:325) internal
at [anon] () native strict preventsyield
at [anon] (function:76) preventsyield

**Comment by ManchM** [comment]
![jack](https://user-images.githubusercontent.com/97548738/152372415-8faabdc4-39c5-45f3-a310-b9522ba6a66e.jpg)

also wen I was trying to import script in azure Zabbix server this is what I'm getting

**Comment by jack-valko** [comment]
Hello. In order to triage this issue properly, I will need access to your appliance.  At the very least, I will need your Meraki API key to try to reproduce the timeout issue on my developer instance.  If you want to arrange this please email me separately at jack at valko dot org.

**Comment by JoRoSouth** [comment]
I'm having something similar.
![image](https://user-images.githubusercontent.com/44182574/153340899-caaf9cf4-88a7-4f74-83bb-4b53320252c4.png)
Cannot execute script: Error: cannot get URL: Timeout was reached.
at [anon] (httprequest.c:336) internal
at [anon] () native strict preventsyield
at [anon] (function:90) preventsyield

Started setting this up and can't go any further on zabbix 5.4.10

**Comment by jack-valko** [comment]
Hi JoRoSouth,
Please confirm you can hit the Meraki Dashboard API from the command line on the zabbix server.  Also, tell me more about the installed environment (packages, sources, cloud image, container, or appliance) and OS (which linux release).
Thank you,
Jack

**Comment by JoRoSouth** [comment]
Debian 10
Zabbix packages from official zabbix repository
 wget https://repo.zabbix.com/zabbix/5.4/debian/pool/main/z/zabbix-release/zabbix-release_5.4-1+debian10_all.deb
# dpkg -i zabbix-release_5.4-1+debian10_all.deb
# apt update


Large reserve of resources for zabbix_server as it is a vm on a proxmox.

```
curl -I -k https://api-mp.meraki.com/
HTTP/1.1 302 Found
Server: nginx/1.21.3
Date: Thu, 10 Feb 2022 23:32:12 GMT
Content-Type: text/html; charset=utf-8
Connection: keep-alive
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: Fri, 01 Jan 1990 00:00:00 GMT
Location: https://account.meraki.com/login/dashboard_login?go=%2F
X-UA-Compatible: IE=Edge,chrome=1
Set-Cookie: _session_id=448aa6caf7ee59fece0789d551fca279; path=/; secure; HttpOnly
X-Request-Id: 9bc51b339188fb502fe87b05382a599b
X-Runtime: 0.009596
Strict-Transport-Security: max-age=15552000; includeSubDomains

```

![image](https://user-images.githubusercontent.com/44182574/153515382-f378c7c6-2ab6-413e-9629-2e7d5a863b73.png)


time curl -I -ks -H 'X-Cisco-Meraki-API-Key: dookdokdookdokreplacedkey' https://api-mp.meraki.com/api/v1/organizations
HTTP/1.1 200 OK
Server: nginx/1.21.3
Date: Thu, 10 Feb 2022 23:44:11 GMT
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Vary: Accept-Encoding
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: Fri, 01 Jan 1990 00:00:00 GMT
X-Frame-Options: sameorigin
X-Robots-Tag: none
X-UA-Compatible: IE=Edge,chrome=1
X-Request-Id: 0acb89c814977485e658e46e5b714101
X-Runtime: 0.033954
Strict-Transport-Security: max-age=15552000; includeSubDomains
X-XSS-Protection: 1; mode=block


real    0m0.872s
user    0m0.009s
sys     0m0.011s

**Comment by jack-valko** [comment]
That is a pretty big Meraki network.  Are all of these devices in an 'online' state?  Do you many that are 'offline' or 'dormant'?

**Comment by jack-valko** [comment]
Here's something to try:
https://www.zabbix.com/documentation/current/en/manual/web_interface/frontend_sections/administration/general#communication-with-zabbix-server

Set the "Network timeout for script execution" value to 300 seconds and try your discovery again.  I doubt this will work, but it's worth a try.

**Comment by JoRoSouth** [comment]
Did that then the same error.  I did change the Meraki device discovery
rule timeout to 60 from 30.  It said timeout exceeded when connecting to
zabbix server "localhost"

**Comment by jack-valko** [comment]
I've opened a bug with Zabbix.  [ZBX-20572](https://support.zabbix.com/browse/ZBX-20572)
I'll look to porting the discovery to an external script for very large networks like yours.

**Comment by JoRoSouth** [comment]
Probably about 100 could be considered dormant depending on deployment with
10+ randomly offline at any given day.

**Comment by JoRoSouth** [comment]
Thank you for your help.

**Comment by gene6482** [comment]
I'm experiencing the same exact issue.  I only have 133 devices.  It worked previously.

**Comment by jack-valko** [comment]
Hi gene6482,

This is a great report, thank you.  Tell me what has changed from previous to current.

Jack

**Comment by gene6482** [comment]
I hadn't noticed it right away, but I believe it may be related to a migration that I did from CentOS to Rocky Linux on Monday morning.  I used the rocky linux migration script and everything appears to be functional otherwise.  It only seems to be affecting this discovery rule.  

For reference, this is the link to the rocky linux documentation for the migration.  https://docs.rockylinux.org/guides/migrate2rocky/

**Comment by jack-valko** [comment]
Make sure you can reach the Meraki Dashboard API from the rockylinux command line.  You might also consider compiling Zabbix from sources.

**Comment by gene6482** [comment]
```
curl --header "X-Cisco-Meraki-API-Key: my-api-key" https://api-mp.meraki.com/api/v1/organizations
curl: (51) SSL: no alternative certificate subject name matches target host name 'api-mp.meraki.com'
```
When I checked the certificate on api-mp.meraki.com, it's actually issued to ios.meraki.com.  If I run the curl with --insecure, it works.

**Comment by gene6482** [comment]
So...I attempted to import the file, had an error about an inventory field.  Commented out.  Another one.  Commented out.  I then went to import it and came upon the actual cause of the error.  Apparently another template that I had installed renamed or changed the ICMP Ping template.  Once I restored that, the discovery worked again.  Thank you so much for your help.  (I know I wasn't the reporter, but maybe it will lead them in the right direction as well)
